File: app/applet/apply_theme.py
import os
import re

def process_class_string(class_str):
    if not class_str:
        return class_str
    
    classes = class_str.split()
    new_classes = []
    has_card = False
    
    for c in classes:
        # Remove rounded
        if c.startswith('rounded-') or 'shadow-' in c:
            continue
        
        # Replace backgrounds
        if c.startswith('bg-[#'):
            new_classes.append('bg-card')
            has_card = True
        elif c == 'border-white/10' or c == 'border-white/5' or c == 'border-white/20':
            new_classes.append('border-border')
            new_classes.append('border-2')
        elif 'text-zinc-' in c:
            new_classes.append('text-muted-foreground')
        elif 'bg-blue-' in c or 'bg-purple-' in c:
            new_classes.append('bg-primary')
            new_classes.append('text-primary-foreground')
        else:
            new_classes.append(c)
            
    if has_card and 'brut-card' not in new_classes:
        new_classes.append('brut-card')
        
    # Deduplicate keeping order
    seen = set()
    final_classes = []
    for c in new_classes:
        if c not in seen:
            seen.add(c)
            final_classes.append(c)
            
    return " ".join(final_classes)

def process_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    # Find className="..."
    def replace_double_quote(match):
        orig_classes = match.group(1)
        new_classes = process_class_string(orig_classes)
        return f'className="{new_classes}"'

    content = re.sub(r'className="([^"]+)"', replace_double_quote, content)
    
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(content)

for root, dirs, files in os.walk('src'):
    for file in files:
        if file.endswith('.tsx'):
            process_file(os.path.join(root, file))

File: app/applet/apply_theme.js
import fs from "fs";
import path from "path";

function processClassString(classStr) {
  if (!classStr) return classStr;
  
  const classes = classStr.split(/\s+/);
  const newClasses = [];
  let hasCard = false;
  
  for (const c of classes) {
    if (c.startsWith("rounded-") || c.includes("shadow-")) continue;
    
    if (c.startsWith("bg-[#")) {
      newClasses.push("bg-card");
      hasCard = true;
    } else if (c === "border-white/10" || c === "border-white/5" || c === "border-white/20") {
      newClasses.push("border-border");
      newClasses.push("border-2");
    } else if (c.startsWith("text-zinc-")) {
      newClasses.push("text-muted-foreground");
    } else if (c.startsWith("bg-blue-") || c.startsWith("bg-purple-") || c.startsWith("bg-emerald-")) {
      newClasses.push("bg-primary");
      newClasses.push("text-primary-foreground");
    } else {
      newClasses.push(c);
    }
  }
  
  if (hasCard && !newClasses.includes("brut-card")) {
    newClasses.push("brut-card");
  }
  
  return [...new Set(newClasses)].join(" ");
}

function processFile(filepath) {
  let content = fs.readFileSync(filepath, "utf-8");
  
  // replace className="..."
  content = content.replace(/className="([^"]+)"/g, (match, p1) => {
    return `className="${processClassString(p1)}"`;
  });
  
  // replace className={`...`}
  // simplified replacement for template literals, won't handle variable interpolation perfectly but will replace static strings inside it
  content = content.replace(/className=\{`([^`]+)`\}/g, (match, p1) => {
    return `className={\`${processClassString(p1)}\`}`;
  });

  fs.writeFileSync(filepath, content, "utf-8");
}

function walk(dir) {
  const files = fs.readdirSync(dir);
  for (const file of files) {
    const p = path.join(dir, file);
    if (fs.statSync(p).isDirectory()) {
      walk(p);
    } else if (p.endsWith(".tsx") || p.endsWith(".ts")) {
      processFile(p);
    }
  }
}

walk("src");
